util/src/lib.rs

```diff
@@ -2,6 +2,10 @@
 //! libraries.
 #![cfg_attr(target_os = "none", no_std)]
 
+#[cfg(feature = "alloc")]
+extern crate alloc;
```

Comment:
I'm guessing this feature will be enabled by our allocator crate when it is added as a dependency?

Reply:
it's enabled by the kernel when it has an allocator and/or by code which requires alloc.